File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def prodOfDigits(self, num):
        prod = 1
        while num:
            prod *= num % 10
            num //= 10
        return prod

    def smallestNumber(self, n: int, t: int) -> int:
        while self.prodOfDigits(n) % t != 0:
            n += 1
        return n

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int prodOfNum(int n){
        int prod = 1;
        while (n > 0){
            int lastDig = n % 10;
            prod = prod * lastDig;
            n = n / 10;
        }
        return prod;
    }

    int smallestNumber(int n, int t) {
        
        while (prodOfNum(n) % t != 0) {
            n++;
        }
        return n;
    }
};